lib/poetry-dep-graph-builder.ts

```diff
@@ -0,0 +1,95 @@
+import { DepGraph, DepGraphBuilder, PkgInfo } from '@snyk/dep-graph';
+import { PoetryLockFileDependency } from './lock-file-parser';
+
+// Poetry uses the virtualenv to create an environment and this comes with these
+// packages pre-installed, therefore they won't be part of the lockfile.
+// See: https://github.com/python-poetry/poetry/issues/3075#issuecomment-703334427
+const IGNORED_DEPENDENCIES: string[] = [
+  'setuptools',
+  'distribute',
+  'pip',
+  'wheel',
+];
+
+export function build(
+  pkgDetails: PkgInfo,
+  pkgDependencyNames: string[],
+  pkgSpecs: PoetryLockFileDependency[],
+): DepGraph {
+  const builder = new DepGraphBuilder({ name: 'poetry' }, pkgDetails);
+  addDependenciesToGraph(
+    pkgDependencyNames,
+    pkgSpecs,
+    builder.rootNodeId,
+    builder,
+  );
+  return builder.build();
+}
+
+export function addDependenciesToGraph(
```

Comment:
why do you export everything?
please only export the main `build` function

Reply:
So I was wondering about this. I'd like to be able to test these functions in isolation, but I don't want them to be exported as usable in the package. You're right, the only thing that needs to be exported is the `build` function and perhaps the error. I guess the equivalent in .NET I'm looking for is `internal`. Is there a way for me to achieve this here?